src/components/Comment/index.spec.tsx

```diff
@@ -1,82 +1,309 @@
+    const root = render({ commentId: comment.id, store, readOnly: false });
 
     expect(root.find(`.${styles.form}`)).toHaveLength(1);
-    expect(root.find(`.${styles.textarea}`)).toHaveProp(
-      'value',
-      comment.content,
+    const textarea = root.find(`.${styles.textarea}`);
+    expect(textarea).toHaveLength(1);
+    expect(textarea).toHaveProp('disabled', false);
+    expect(textarea).toHaveProp('value', comment.comment);
+
+    const button = root.find(Button);
+    expect(button).toHaveProp('disabled', false);
+    expect(button).toHaveText('Save');
+  });
+
+  it('focuses on mount when a textarea ref exists', () => {
+    const focus = jest.fn();
+    render({ createTextareaRef: () => createFakeTextareaRef({ focus }) });
+
+    expect(focus).toHaveBeenCalled();
+  });
+
+  it('submits a new comment', () => {
+    const addonId = 222;
+    const fileName = 'manifest.json';
+    const line = 432;
+    const versionId = 321;
+    const store = configureStore();
+    const dispatchSpy = spyOn(store, 'dispatch');
+
+    const fakeThunk = createFakeThunk();
+    const _manageComment = fakeThunk.createThunk;
+
+    const commentText = 'Example of a comment';
+
+    const root = render({
+      _manageComment,
+      addonId,
+      fileName,
+      line,
+      store,
+      versionId,
+    });
+
+    root.find(`.${styles.textarea}`).simulate(
+      'change',
+      createFakeChangeEvent({
+        name: 'input',
+        value: commentText,
+      }),
     );
+
+    root.find(`.${styles.form}`).simulate('submit', createFakeEvent());
+
+    expect(dispatchSpy).toHaveBeenCalledWith(fakeThunk.thunk);
+    expect(_manageComment).toHaveBeenCalledWith({
+      addonId,
+      cannedResponseId: undefined,
+      comment: commentText,
+      commentId: undefined,
+      fileName,
+      line,
+      versionId,
+    });
   });
 
-  describe('keydown protection', () => {
-    const createFakeTextareaRef = () => {
-      return {
-        ...React.createRef(),
-        current: {
-          ...document.createElement('textarea'),
-          addEventListener: jest.fn(),
-          removeEventListener: jest.fn(),
-        },
-      };
-    };
+  it('renders a pending state while saving a comment', () => {
+    const keyParams = { fileName: null, line: null, versionId: 1 };
+    const pendingCommentText = 'Example of a comment being edited';
+    const store = configureStore();
+    store.dispatch(
+      actions.beginSaveComment({ ...keyParams, pendingCommentText }),
+    );
+
+    const root = render({ readOnly: false, store, ...keyParams });
 
+    const textarea = root.find(`.${styles.textarea}`);
+    expect(textarea).toHaveProp('disabled', true);
+    expect(textarea).toHaveProp('value', pendingCommentText);
+
+    const button = root.find(Button);
+    expect(button).toHaveProp('disabled', true);
+    expect(button).toHaveText('Saving…');
+  });
+
+  it('seeds the form with an initial comment', () => {
```

Comment:
I find a bit of a disconnect between this test's description and what it actually checks. From the description I would have expected it to check the contents of the textarea, but what it actually does is check what is submitted when the form is submitted. Would the former be a more accurate test for this, or should the description of the test perhaps be changed?

Reply:
I didn't check the textarea value because it seemed like I was reaching into the implementation too much. However, a textarea is how you make a form, heh. I guess it's fine to check the value. The test was taking a previously saved value and checking that it was submitted as part of the form to imply that this value was used in the textarea.